h/accounts/models.py

```diff
@@ -132,6 +133,11 @@ def is_activated(self):
 
         return False
 
+    def activate(self, request):
+        """Activate the user by deleting any activation they have."""
+        request.db.delete(self.activation)
+        request.registry.notify(events.ActivationEvent(request, self))
```

Comment:
We're talking about duplication of a trivial piece of code in two places. It's not even a disaster if someone forgets to call it. I'd much rather have duplication than the wrong abstraction, and having a model class which accepts `request` objects is definitely the wrong abstraction :).